File: demos/47711/σ░ÅΦ▒åσ¡É/frontend/api/activity.js
import { request } from '../utils/api'

const PREFIX = '/activities'

function normalizeActivitySignup(item = {}) {
	return {
		id: item.id,
		userId: item.user_id ?? null,
		name: item.name || '',
		phone: item.phone || '',
		note: item.note || '',
		signupTime: item.signup_time ? new Date(item.signup_time).getTime() : 0
	}
}

function normalizeActivity(item = {}) {
	return {
		id: item.id,
		emoji: item.emoji || '🎭',
		title: item.title || '',
		description: item.description || '',
		date: item.date || '',
		time: item.time || '',
		location: item.location || '',
		maxParticipants: item.max_participants || 0,
		tags: Array.isArray(item.tags) ? item.tags : [],
		organizer: item.organizer || '',
		organizerAvatar: item.organizer_avatar || '',
		status: item.status || '报名中',
		signupCount: item.signup_count || 0,
		isJoined: !!item.is_signed_up,
		mySignupTime: item.my_signup_time ? new Date(item.my_signup_time).getTime() : 0,
		signups: Array.isArray(item.signups) ? item.signups.map(normalizeActivitySignup) : []
	}
}

function extractList(res) {
	const list = Array.isArray(res?.data) ? res.data : []
	return list.map(normalizeActivity)
}

function extractDetail(res) {
	return normalizeActivity(res?.data || res)
}

export function activityRequest({ url = '', method = 'GET', data = null, auth = false, header = {} } = {}) {
	return request({ url: `${PREFIX}${url}`, method, data, auth, header })
}

export async function listActivities() {
	const res = await activityRequest({ auth: true })
	return extractList(res)
}

export async function getActivityDetail(id) {
	const res = await activityRequest({ url: `/${id}`, auth: true })
	return extractDetail(res)
}

export async function createActivitySignup(id, data) {
	const res = await activityRequest({ url: `/${id}/signup`, method: 'POST', data, auth: true })
	return extractDetail(res)
}

export async function deleteActivitySignup(id) {
	const res = await activityRequest({ url: `/${id}/signup`, method: 'DELETE', auth: true })
	return extractDetail(res)
}

export async function listMyActivitySignups() {
	const res = await activityRequest({ url: '/my-signups', auth: true })
	return extractList(res)
}

File: demos/47711/σ░ÅΦ▒åσ¡É/backend/app/schemas/activity.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field, field_validator


class ActivitySignupRequest(BaseModel):
    name: str = Field(min_length=1, max_length=30)
    phone: str = Field(min_length=11, max_length=11)
    note: str | None = Field(default=None, max_length=200)

    @field_validator("name")
    @classmethod
    def validate_name(cls, value: str) -> str:
        cleaned = value.strip()
        if not cleaned:
            raise ValueError("姓名不能为空")
        return cleaned

    @field_validator("phone")
    @classmethod
    def validate_phone(cls, value: str) -> str:
        if not value.isdigit() or len(value) != 11:
            raise ValueError("请输入11位手机号")
        return value

    @field_validator("note")
    @classmethod
    def validate_note(cls, value: str | None) -> str | None:
        if value is None:
            return None
        cleaned = value.strip()
        return cleaned or None


class ActivitySignupOut(BaseModel):
    id: int
    user_id: int | None = None
    name: str
    phone: str
    note: str | None = None
    signup_time: datetime

    model_config = ConfigDict(from_attributes=True)


class ActivityOut(BaseModel):
    id: int
    emoji: str
    title: str
    description: str
    date: str
    time: str
    location: str
    max_participants: int
    tags: list[str]
    organizer: str
    organizer_avatar: str | None = None
    status: str
    signup_count: int
    is_signed_up: bool = False
    my_signup_time: datetime | None = None
    signups: list[ActivitySignupOut] = Field(default_factory=list)
